app/models/order_item.rb

```diff
@@ -40,10 +40,10 @@ def set_defaults
   end
 
   def update_message(level, message)
-    self.updated_at = Time.zone.now
-    ProgressMessage.create(:level         => level,
-                           :message       => message,
-                           :order_item_id => id)
+    new_message = ProgressMessage.new(:level => level, :message => message)
+    progress_messages << new_message
+    touch
+    save!
```

Comment:
We do not need `new_message` anymore and I think the `save!` is also not needed.

```ruby
progress_messages << ProgressMessage.new(:level => level, :message => message)
touch
```